core/logging_handlers.py

```diff
@@ -0,0 +1,68 @@
+from core.rovecomm import RoveCommPacket
+import core
+import logging
+from datetime import datetime
+import os
+
+
+class CsvHandler(logging.handlers.WatchedFileHandler):
+
+    def __init__(self, filename, format_string, encoding=None, delay=False, new_file=False):
+        """
+        Initializes the handler
+        """
+
+        if (new_file):
+            timestamp = datetime.now().strftime("%Y%m%d-%H%M%S")
+            if not os.path.exists(
+                "/".join(
+                    f'{filename[:-4]}-{timestamp}{filename[-4:]}'.split("/")[:-1]
+                )
+            ):
+                os.makedirs("/".join(f'{filename[:-4]}-{timestamp}{filename[-4:]}'.split("/")[:-1]))
+            f = open(f'{filename[:-4]}-{timestamp}{filename[-4:]}', 'w')
+            f.write(format_string + '\n')
+            f.close()
+            logging.handlers.WatchedFileHandler.__init__(
+                self,
+                f'{filename[:-4]}-{timestamp}{filename[-4:]}',
+                'a',
+                encoding,
+                delay
+            )
+        else:
+            if (not os.path.exists(filename)):
+                if not os.path.exists("/".join(filename.split("/")[:-1])):
+                    os.makedirs("/".join(filename.split("/")[:-1]))
+                f = open(filename, 'w')
+                f.write(format_string + '\n')
+                f.close()
+            logging.handlers.WatchedFileHandler.__init__(self, filename, 'a', encoding, delay)
+
+
+class RoveCommHandler(logging.Handler):
+    def __init__(self, reliable):
+        """
+        Initializes the handler with given network variables
+        """
+        self.reliable = reliable
+
+        logging.Handler.__init__(self)
+
+    def emit(self, s):
+        """
+        Encodes and sends the log message over RoveComm
+        """
+        msg = self.format(s)
+        # Max string size is 255 characters, truncate the rest and flag it
+        if len(msg) > 255:
+            msg = msg[:252] + "..."
+
+        packet = RoveCommPacket(
+            4241,
```

Comment:
make dataid a config in the yaml file rather than hardcoded in this obscure place